chainer/functions/loss/discriminative_loss.py

```diff
@@ -0,0 +1,248 @@
+import cupy as cp
```

Comment:
We would like to write Chainer functions in CPU/GPU-agnostic way. When users run code in CPU, `cupy` is not needed (and is typically not unavailable). One typical way to check in code is to use [`get_array_module`](https://docs.chainer.org/en/v4/reference/util/generated/chainer.backends.cuda.get_array_module.html). See its usage e.g. [here](https://github.com/chainer/chainer/blob/3ba7d4d0ddcc0202225161f75e6d6fed07ffa25e/chainer/functions/math/ceil.py).